### PR TITLE
Add browser auth integration test

### LINKED ISSUE #101: Create tests for `@itwin/browser-authorization`
The `@itwin/browser-authorization` package doesn't have any tests that test authorization, so create some.

At the very least, create tests for the main functionality of the package, i.e., that signing in and  signing out works.

Probably can code the tests similar to the the tests in [oidc-signin-tool](https://github.com/iTwin/auth-clients/tree/main/packages/oidc-signin-tool/src).

### PR DESCRIPTION
**Pull request by ben-polinsky**
Add playwright tests for Browser based authentication, both via redirect and via popup. 

Closes #101

### CODE REVIEW

**Comment by ben-polinsky** [comment]
I believe I can refine the playwright binaries are downloaded...

**Comment by ben-polinsky** [comment]
Will also see if I can recreate #92 and add a test for this

**Comment by ben-polinsky** [comment]
@aruniverse not a fan of this nonblock-statement-body-position lint rule...

File: packages/browser/package.json [hunk 1]
```diff
@@ -44,7 +46,13 @@
     "sinon": "^15.0.1",
     "eslint": "^7.32.0",
     "rimraf": "^3.0.2",
-    "typescript": "~4.3.5"
+    "typescript": "~4.3.5",
+    "@playwright/test": "~1.31.1",
```
Comment (aruniverse) [inline packages/browser/package.json:50]: could you alphabetize the deps?

File: packages/browser/src/integration-tests/test-app/index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (aruniverse) [inline packages/browser/src/integration-tests/test-app/index.html:1]: copyright?

File: packages/browser/package.json [hunk 3]
```diff
@@ -13,6 +13,8 @@
     "lint": "eslint -f visualstudio \"./src/**/*.ts\" 1>&2",
     "lint:fix": "npm run lint -- --fix",
     "test": "mocha \"./lib/cjs/test/**/*.test.js\"",
+    "test:integration:start-test-app": "parcel src/integration-tests/test-app/index.html",
```
Comment (aruniverse) [inline packages/browser/package.json:16]: can you update the npmignore file, to make sure we dont accidentally deliver any of these files?

File: packages/browser/package.json [hunk 4]
```diff
@@ -44,7 +46,13 @@
     "sinon": "^15.0.1",
     "eslint": "^7.32.0",
     "rimraf": "^3.0.2",
-    "typescript": "~4.3.5"
+    "typescript": "~4.3.5",
+    "@playwright/test": "~1.31.1",
+    "http-server": "~14.1.1",
```
Comment (aruniverse) [inline packages/browser/package.json:51]: do you need http-server or buffer?

**Review by aruniverse — COMMENTED** [review]
youre the mvp! overall looks good, few comments:

- you might need to rebase to update the first few commits, leaking your personal account.
- can you update the readme with instructions on how to run the tests?
- can you update the npmignore to make sure we dont accidentally deliver these test files?

File: packages/browser/package.json (hunk 4 above)
Comment (ben-polinsky) [reply packages/browser/package.json:51]: Buffer is needed as a polyfill for parcel - which is a dream to setup compared to webpack. http-server is left over from initial testing - will remove.

File: packages/browser/package.json (hunk 3 above)
Comment (ben-polinsky) [reply packages/browser/package.json:16]: done - also excluded unit tests

File: packages/browser/src/integration-tests/test-app/index.html (hunk 2 above)
Comment (ben-polinsky) [reply packages/browser/src/integration-tests/test-app/index.html:5]: Didn't know we needed 'em for test html files, but added.

File: packages/browser/package.json (hunk 1 above)
Comment (ben-polinsky) [reply packages/browser/package.json]: done

File: packages/browser/package.json [hunk 5]
```diff
@@ -13,6 +13,8 @@
     "lint": "eslint -f visualstudio \"./src/**/*.ts\" 1>&2",
     "lint:fix": "npm run lint -- --fix",
     "test": "mocha \"./lib/cjs/test/**/*.test.js\"",
+    "test:integration:start-test-app": "parcel src/integration-tests/test-app/index.html",
+    "test:integration": "playwright test src/integration-tests",
```
Comment (aruniverse) [inline packages/browser/package.json:17]: could you add the `test-results` folder created to the gitignore?

Comment (ben-polinsky) [reply packages/browser/package.json:17]: Already there

Comment (aruniverse) [reply packages/browser/package.json:17]: ah my b, its there. my vscode instance didnt highlight the folder so i thought it wasnt ignored

**Review by aruniverse — APPROVED** [review]

**Pull request merged**